pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -40,22 +47,59 @@ let
     # lua code to put into the generated init.lua file
     , luaRcContent ? ""
     # entry to load in packpath
-    , packpathDirs
+    , packpathDirs ? null # not used anymore
+    , plugins ? []
     , ...
-  }:
+  }@attrs:
   assert withPython2 -> throw "Python2 support has been removed from the neovim wrapper, please remove withPython2 and python2Env.";
 
+  # assert packpathDirs -> throw "Packpathdirs removed";
+
   stdenv.mkDerivation (finalAttrs:
   let
+    pluginsNormalized = neovimUtils.normalizePlugins plugins;
+
+    myVimPackage = neovimUtils.normalizedPluginsToVimPackage pluginsNormalized;
+
+    rubyEnv = bundlerEnv {
+      name = "neovim-ruby-env";
+      gemdir = ./ruby_provider;
+      postBuild = ''
+        ln -sf ${ruby}/bin/* $out/bin
+      '';
+    };
+
+    pluginRC = lib.foldl (acc: p: if p.config != null then acc ++ [p.config] else acc) []  pluginsNormalized;
+
+    manifestRc = vimUtils.vimrcContent { customRC = ""; };
+    # we call vimrcContent without 'packages' to avoid the init.vim generation
+    neovimRcContent' = vimUtils.vimrcContent {
+      beforePlugins = "";
+      customRC = lib.concatStringsSep "\n" (pluginRC ++ [neovimRcContent]);
+      packages = null;
+    };
 
     finalPackdir = neovimUtils.packDir packpathDirs;
 
     rcContent = ''
       ${luaRcContent}
-    '' + lib.optionalString (!isNull neovimRcContent) ''
-      vim.cmd.source "${writeText "init.vim" neovimRcContent}"
+    '' + lib.optionalString (!isNull neovimRcContent') ''
+      vim.cmd.source "${writeText "init.vim" neovimRcContent'}"
     '';
 
+    getDeps = attrname: map (plugin: plugin.${attrname} or (_: [ ]));
+
+    requiredPlugins = vimUtils.requiredPluginsForPackage myVimPackage;
+    pluginPython3Packages = getDeps "python3Dependencies" requiredPlugins;
+
+    python3Env = lib.warnIf (attrs ? python3Env) "Pass your python packages via the `extraPython3Packages`, e.g., `extraPython3Packages = ps: [ ps.pandas ]`"
+      python3.pkgs.python.withPackages (ps:
+      [ ps.pynvim ]
+      ++ (extraPython3Packages ps)
+      ++ (lib.concatMap (f: f ps) pluginPython3Packages));
+
+    packpathDirs.myNeovimPackages = myVimPackage;
+
     wrapperArgsStr = if lib.isString wrapperArgs then wrapperArgs else lib.escapeShellArgs wrapperArgs;
```

Comment:
Would it be useful to also have a `runtimeInputs` or `dependencies` argument that prepends to PATH via the wrapper?

Reply:
totally. This will be unblocked by this PR. I prefer it to be done as a follow up though since neovim is a touchy derivation: you quickly know when you break something xD and this has the potential to break configs already (at least for people using the "private/unstable" APIs)